omnibus/files/private-chef-cookbooks/private-chef/templates/default/nginx/nginx_chef_api_lb.conf.erb

```diff
@@ -57,10 +66,13 @@
     <% end -%>
 
     <% if node['private_chef']['data_collector']['root_url'] && node['private_chef']['data_collector']['token'] -%>
-    location ~ /organizations/([^/]+)/data-collector {
+    location ~ "^/organizations/([^/]+)/data-collector$" {
       set $request_org $1;
-      access_by_lua_file '<%= @script_path %>/check_token_using_endpoint.lua';
-      proxy_pass $upstream;
+      access_by_lua_block { validator.validate("POST") }
+      proxy_set_header x-data-collector-token $data_collector_token;
+      proxy_set_header x-data-collector-auth "version=1.0";
+      rewrite ^ <%= URI.parse(node['private_chef']['data_collector']['root_url']).path %> break;
+      proxy_pass <%= URI.parse(node['private_chef']['data_collector']['root_url']).scheme %>://data-collector;
```

Comment:
When the user configures their upstream data-collector we _dont_ enforce https, therefore the scheme changes depending on whether it's https or not. Since the upstream data collector (usually Automate) and Chef server can have (and in most cases do have) different SSL config, I think this is pretty much unavoidable.